### ISSUE TITLE
When onlyif and bg are used together the

### ISSUE THREAD

**Issue by ruiaylin** [saltstack/salt#41626]
### Description of Issue/Question
I want to make a command to start a process background like daemon . 
like this below i want to start a mysql database : 
```
mysql_runing:
    cmd.run:
    - names:
        - /usr/bin/mysqld_safe --defaults-file= ...xxxxx/my.cnf --user=mysql 
    - runas: mysql
    - bg: True

```

It works very well , but i want to do some check before execute the mysqld_safe command . 
My code like this now: 

```
mysql_runing:
    cmd.run:
    - names:
        - /usr/bin/mysqld_safe --defaults-file=...xxxxx/my.cnf --user=mysql
    - onlyif: test -z  `ps -ef|grep -w  mysqld | grep 3306  | grep -v 'grep' | awk -F ' '  '{ print $8 }'  `
    - runas: mysql
    - bg: True

```
This mean that  only if the process not exist , to start it . 

when i run , i get the wrong message like this : 

```
          ID: mysql_runing
    Function: cmd.run
        Name: /usr/bin/mysqld_safe --defaults-file=...xxxxx/my.cnf --user=mysql
      Result: True
     Comment: onlyif execution failed
     Started: 18:22:16.331441
    Duration: 163.719 ms
     Changes:
```

I check the salt minion's log : 
```
[INFO    ][32509] Executing command 'test -z  `ps -ef|grep -w  mysqld | grep 3307  | grep -v 'grep' | awk -F ' '  '{ print $8 }'  `' as user 'mysql' in directory '/home/mysql'. Executing command in the background, no output will be logged.
[DEBUG   ][32509] output: None
[DEBUG   ][32509] Last command return code: None
[INFO    ][32509] onlyif execution failed

```

This mean that if i set the bg variables to True , the onlyif check command execute without output .  Sounds so bad .  If i remove the bg variable , the onlyif checker  execute all right as i thought . 

Is this a Bug ?  If it's not , how can i implement  it ? 

### Versions Report
(Provided by running `salt --versions-report`. Please also mention any differences in master/minion versions.)
```
Salt Version:
           Salt: 2016.11.5

Dependency Versions:
           cffi: Not Installed
       cherrypy: unknown
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.8.1
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.20.2
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: 1.2.5
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: 3.4.3
         pygit2: Not Installed
         Python: 2.6.6 (r266:84292, Aug 18 2016, 15:13:37)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.5.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5

System Versions:
           dist: centos 6.9 Final
        machine: x86_64
        release: 2.6.32-431.el6.x86_64
         system: Linux
        version: CentOS 6.9 Final
```

**Comment by gtmanfred** [comment]
Thanks for reporting this.

I have fixed the problem in #41637

Daniel

**Comment by gtmanfred** [comment]
Closing because the above PR has been merged.

Thanks,
Daniel
